src/osd/PG.cc

```diff
@@ -3820,10 +3820,11 @@ void PG::scrub(ThreadPool::TPHandle &handle)
 	continue;
       if (acting[i] == CRUSH_ITEM_NONE)
 	continue;
-      ConnectionRef con = osd->get_con_osd_cluster(acting[i], get_osdmap()->get_epoch());
+      uint64_t features;
+      ConnectionRef con = osd->get_con_osd_cluster(acting[i], get_osdmap()->get_epoch(), &features);
       if (!con)
 	continue;
-      if (!con->has_feature(CEPH_FEATURE_CHUNKY_SCRUB)) {
+      if (!(features & CEPH_FEATURE_CHUNKY_SCRUB)) {
```

Comment:
here, i think we should just use peer_features.

(and verify that the peer_features bitmask construction uses your improved helper properly)